git_commit.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+import os
+import sh
+import sys
+import json
+import rc_util
+from rc_rmq import RCRMQ
+
+task = 'git_commit'
+
+# Instantiate rabbitmq object
+rc_rmq = RCRMQ({'exchange': 'RegUsr', 'exchange_type': 'topic'})
+
+# Define some location
+repo_location = os.path.expanduser('~/git/rc-users')
+users_dir = repo_location + '/users'
+groups_dir = repo_location + '/groups'
+
+args = rc_util.get_args()
+logger = rc_util.get_logger(args)
+
+if not args.dry_run:
+    git = sh.git.bake('-C', repo_location)
+    ldapsearch = sh.Command('ldapsearch')
+else:
+    git = sh.echo.bake('git', '-C', repo_location)
+    ldapsearch = sh.echo.bake('ldapsearch')
+
+def git_commit(ch, method, properties, body):
+    msg = json.loads(body)
+    username = msg['username']
+    ticketnum = msg.get('ticketnum', 'add-users-' + username.lower())
+    msg['task'] = task
+    msg['success'] = False
+    branch_name = 'issue-' + ticketnum
+    user_ldif = users_dir + f'/{username}.ldif'
+    group_ldif = groups_dir + f'/{username}.ldif'
+
+    logger.info("Received: %s", msg)
+    logger.debug("ticketnum: %s", ticketnum)
+    logger.debug("branch_name: %s", branch_name)
+
+    try:
+
+        logger.debug('git checkout master')
+        git.checkout('master')
+        logger.debug('git pull')
+        git.pull()
+        logger.debug('git checkout -b %s', branch_name)
+        git.checkout('-b', branch_name)
+
+        logger.debug("open(%s, 'w'), open(%s, 'w')", user_ldif, group_ldif)
+        with open(user_ldif, 'w') as ldif_u,\
+            open(group_ldif, 'w') as ldif_g:
+            logger.debug(f"ldapsearch -LLL -x -h ldapserver -b 'dc=cm,dc=cluster' uid={username} > {user_ldif}")
+            ldapsearch('-LLL', '-x', '-h', 'ldapserver', '-b', "'dc=cm,dc=cluster'", f"uid={username}", _out=ldif_u)
```

Comment:
Could you change this `"'dc=cm,dc=cluster'"` to `'dc=cm,dc=cluster'` It is complaining about the double quotes.

Reply:
fixed in d75c72d